cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -88,9 +126,7 @@ def _value_equality_values_(self):
 
     def equal_up_to_global_phase(self, other):
```

Comment:
This is not correct as you are not checking for equality of qubits in here (only checking for equality of underlying gates). Qubit equality was implicitly checked earlier when comparing `pauli_string`s.

We could have used the implicit version defined in `GateOperation` if it used `self._group_interchangeable_qubits` to compare the qubits instead of directly using `self.qubits` (see #4766)

Reply:
Nice catch!

Reply:
Will implement the equivalent qubits function instead.

Reply:
I tried, but it didn't seem to work. We can revisit this later if desired.